src/js/stores/AppsStore.js

```diff
@@ -147,29 +148,47 @@ function processApp(app) {
   return app;
 }
 
-function processEmptyGroup(group) {
-  return Object.assign(processApp({id: group.id}), {
-    isGroup: true,
-    status: null
+function addData(parent, child) {
+  Object.keys(parent).forEach(key => {
+    if (key === "health") {
+      parent[key].forEach((value, index) => {
+        value.quantity += child[key][index].quantity;
+      });
+      return;
+    }
+    parent[key] += child[key];
   });
 }
 
 // TODO: refactor to adhere to /groups endpoint structures.
 // https://github.com/mesosphere/marathon/issues/3383
-function processAppsAndEmptyGroups(group, apps = []) {
-  if (group.apps != null && group.apps.length > 0) {
-    group.apps.forEach(app => {
-      apps.push(processApp(app));
-    });
-  } else if (group.id !== "/") {
-    apps.push(processEmptyGroup(group));
+function processAppsAndGroups(group, root = true) {
+  var apps = [];
+  var data = Util.deepCopy(groupScheme);
```

Comment:
Please update the group scheme as it should include `id` and `isGroup` and inline the the id assignment as proposed:

```
var groupData = Object.assign({}, Util.deepCopy(groupScheme), {id: group.id});
```